fem/quadinterpolator.cpp

```diff
@@ -420,6 +420,34 @@ void QuadratureInterpolator::MultTranspose(
    MFEM_ABORT("this method is not implemented yet");
 }
 
+static void D2QValues1D(const int NE,
+                        const Array<double> &b_,
+                        const Vector &x_,
+                        Vector &y_,
+                        const int vdim = 1,
+                        const int d1d = 0,
+                        const int q1d = 0)
+{
+   auto b = Reshape(b_.Read(), q1d, d1d);
+   auto x = Reshape(x_.Read(), d1d, vdim, NE);
+   auto y = Reshape(y_.Write(), vdim, q1d, NE);
+
+   MFEM_FORALL(e, NE,
+   {
+      for (int c = 0; c < vdim; c++)
+      {
+         for (int q = 0; q < q1d; ++q)
+         {
+            y(c, q, e) = 0.0;
+            for (int d = 0; d < d1d; ++d)
+            {
+               y(c, q, e) += b(q, d) * x(d, c, e);
+            }
```

Comment:
```suggestion
            double val = 0.0;
            for (int d = 0; d < d1d; ++d)
            {
               val += b(q, d) * x(d, c, e);
            }
            y(c, q, e) += val;
```
I would suggests this to avoid writing each time in global memory.

Reply:
Done.